### PR TITLE
TypeScript improvements, MarkerComponent and some fixes

### PR DESCRIPTION
**Pull request by jankrassnigg**

### CODE REVIEW

File: Code/Engine/GameEngine/Gameplay/Implementation/MarkerComponent.cpp [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  if (!m_sMarkerType.IsEmpty())
+  {
+    m_SpatialCategory = ezSpatialData::RegisterCategory(m_sMarkerType.GetString());
+  }
+  else
+  {
+    m_SpatialCategory = ezInvalidSpatialDataCategory;
+  }
+
+  if (GetOwner())
```
Comment (C-Core) [inline Code/Engine/GameEngine/Gameplay/Implementation/MarkerComponent.cpp]: if (IsActiveAndInitialized())

File: Code/Engine/GameEngine/Gameplay/Implementation/MarkerComponent.cpp [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+void ezMarkerComponent::DeserializeComponent(ezWorldReader& stream)
+{
+  SUPER::DeserializeComponent(stream);
+  // const ezUInt32 uiVersion = stream.GetComponentTypeVersion(GetStaticRTTI());
+  auto& s = stream.GetStream();
+
+  s >> m_sMarkerType;
+  s >> m_fRadius;
+
+  UpdateMarker();
```
Comment (C-Core) [inline Code/Engine/GameEngine/Gameplay/Implementation/MarkerComponent.cpp]: Not necessary to call UpdateMarker here. It won't do anything. The component is not initialized yet so it won't receive the updatelocalbounds message.

File: Code/Engine/GameEngine/Gameplay/Implementation/MarkerComponent.cpp [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+}
+
+void ezMarkerComponent::OnDeactivated()
+{
+  SUPER::OnDeactivated();
+
+  // TODO: do this ?
+  GetOwner()->UpdateLocalBounds();
```
Comment (C-Core) [inline Code/Engine/GameEngine/Gameplay/Implementation/MarkerComponent.cpp]: Yes you need to do this here so the game object's bounds and spatial data are updated excluding this component.

File: Data/Samples/Testing Chambers/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{
+  "compilerOptions": {
+    "target": "es5",
+    "baseUrl": "",
+    "paths": {
+      "*": ["D:/GitHub/ezEngine/Data/Plugins/TypeScript/*", "D:/GitHub/ezEngine/Data/Samples/Testing Chambers/*"]
```
Comment (C-Core) [inline Data/Samples/Testing Chambers/tsconfig.json]: Absolute path doesn't make sense here?

Comment (jankrassnigg) [reply Data/Samples/Testing Chambers/tsconfig.json]: WIP, will change soon

**Pull request merged**